src/core/qgsvectorlayerrenderer.cpp

```diff
@@ -56,7 +57,9 @@ QgsVectorLayerRenderer::QgsVectorLayerRenderer( QgsVectorLayer *layer, QgsRender
   , mLabelProvider( nullptr )
   , mDiagramProvider( nullptr )
 {
+  mContext.distanceArea().setSourceCrs( QgsProject::instance()->crs() );
```

Comment:
remove this - it's not required as it's QgsRenderContext::fromMapSettings 's responsibility to setup the distance area..